### ISSUE TITLE
[ macOs - Asp.net core 1.1 ] Error at startup if node-gyp is on node-modules

### ISSUE THREAD

**Issue by AlexTeixeira** [nodejs/node-gyp#1210]
Hello guys,

Since yesterday, I configurer my Asp.net core with Webpack and some loader in macOs. Sass-loader or node-sass seems to have a dependency to node-gyp so it was it is present in my node modules.

Firstly, after the package installed I get this errors in visual studio code :
<img width="854" alt="capture d ecran 2017-05-30 a 21 13 42" src="https://cloud.githubusercontent.com/assets/6555080/26600590/ededcd6a-457c-11e7-971b-927294299470.png">

That just warning so I didn't investigate a lot on them.

After all my configuration done, I start my application and I got this error : 

<img width="1271" alt="capture d ecran 2017-05-30 a 21 15 13" src="https://cloud.githubusercontent.com/assets/6555080/26600633/194ae88a-457d-11e7-87bf-9dbd5a2ba8f1.png">

This issue did not block me because I juste have to delete the folder to make work my application but every time the node module are reload or updated, I need to delete the node-gyp folder.

I create here an issue, if is not your responsibility I will create a issue for asp.net core team.

Thanks

**Comment by bnoordhuis** [comment]
It's not wholly clear to me what you are trying to do.  node-gyp is a build tool and one of its components is Find-VS2017.cs.  You are not supposed to invoke it directly.

**Comment by AlexTeixeira** [comment]
I'm not invoking it. The package is on my node_modules folder and an exception as raised by asp.net core.

**Comment by bnoordhuis** [comment]
Let me rephrase: node-gyp is a script that you invoke from the command line to build add-ons (native modules) for node.js.  You should not be building it as part of your project.

**Comment by AlexTeixeira** [comment]
Ok. So it's why I can delete without breaking anything. 

So for this case what I should do :

- continue deleting the folder to run my app?
- open an issue in asp.net core ?
- Something else...

**Comment by bnoordhuis** [comment]
The module that depends on node-gyp should probably remove it from its dependencies.  There is rarely a reason to take a direct dependency on node-gyp and in fact it's something of an anti-pattern.

I'll go ahead and close this out as it's not a node-gyp issue.

**Comment by refack** [comment]
> I'll go ahead and close this out as it's not a node-gyp issue.

There is something interesting here. `Find-VS2017.cs` is behind an `if (process.platform == 'win32')` so it should not be run on macOS.

@AlexTeixeira you said you are running it on macOS?

**Comment by bnoordhuis** [comment]
node-gyp isn't run.  VS Code rounds up everything ending in .cs and packages that up in an assembly.

**Comment by AlexTeixeira** [comment]
@refack Yes I'm running my app using VS Code in macOS.

@bnoordhuis I will have a look to the dependencies of my packages and I will do the necessary. Thanks for your time and your explanations.
